ivy/functional/frontends/jax/lax/control_flow_operators.py

```diff
@@ -58,3 +58,21 @@ def while_loop(cond_fun, body_fun, init_val):
     while cond_fun(val):
         val = body_fun(val)
     return val
+
+
+@to_ivy_arrays_and_back
+def scan(f, init, xs, length=None, reverse=False, unroll=1):
```

Comment:
I think the way you are using 'unroll' and 'reverse' might be incorrect. Would recommend looking through the jax docs/source code to help solve this.
https://jax.readthedocs.io/en/latest/_autosummary/jax.lax.scan.html
https://jax.readthedocs.io/en/latest/_modules/jax/_src/lax/control_flow/loops.html#scan